internal/pkg/agent/cmd/enroll.go

```diff
@@ -478,6 +480,12 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command) error {
 		fixPermissions = nil
 	}
 
+	// If agent was installed via a deb or rpm package, we want to avoid restarting because it may not be running yet.
+	// In this case, the user is expected to (re)start agent via systemd instead.
+	if pkgmgr.InstalledViaExternalPkgMgr() {
+		skipDaemonReload = true
+	}
```

Comment:
Why is this only for external package manager? Wouldn't it be good to make this change for all installation types?